fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordValidationTest.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * RecordValidationTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.record.TestRecords1Proto;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.BooleanSource;
+import com.google.common.base.Strings;
+import com.google.protobuf.Message;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.annotation.Nonnull;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class RecordValidationTest extends FDBRecordStoreTestBase {
```

Comment:
Would it make sense for this to have a randomized test that deletes a random subset of the keys for the record? If you only save one record it could delete a random key, or subrange from the entire record subspace.

Reply:
Done (not randomized - scan through all the possible split combinations)

Reply:
Done.

Reply:
Done.